packages/jaeger-ui/src/utils/ant-design.tsx

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2020 The Jaeger Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DefaultOptionType } from 'antd/lib/select';
+
+/**
+ * This enables search by label in Ant Design's Select component (v4.x)
+ *
+ * Pass this to the `filterOption` prop with `showSearch` prop set to `true`
+ *
+ * @example
+ * ```
+ * <Select {...props} filterOption={antSelectSearchByLabel} showSearch={true} />
+ * ```
+ */
+// eslint-disable-next-line import/prefer-default-export
+export function antSelectSearchByLabel(input: string, option?: DefaultOptionType) {
+  return (option?.children?.toString() ?? '').toLowerCase().includes(input.toLowerCase());
```

Comment:
I am not following this logic. What is the expectation of `filterOption` function? Isn't it to return true/false if a given option matches the search? If so, shouldn't the checking be done against `option.value` rather than `option.children`?

Reply:
That's how the default behaviour works, it searches based on `option.value` but more often than not `option.value` is different from what is displayed on the UI through the `option.label` value

For e.g. In the `SearchForm`, lookback dropdown have the labels in human-readable form like `5 Minutes` while internally it used the value `5m`, the user doesn't know about the internal value, they would search by labels displayed to them

<img width="261" alt="Screenshot 2023-09-20 at 2 16 05 AM" src="https://github.com/jaegertracing/jaeger-ui/assets/28570857/ba2b05cc-5ada-4730-8e44-aac01aadb4d0">

The `DefaultOptionType` doesn't have the `label` directly accessible, it rather has `children` property that contains the label, this is the reason I combined the string literal `'Last'` and `label` value so that it becomes a single child instead of both being different children, slight performance improvement
 
<img width="572" alt="Screenshot 2023-09-20 at 2 25 19 AM" src="https://github.com/jaegertracing/jaeger-ui/assets/28570857/cffb598f-7414-4cec-ba50-9f1dc141b786">